eegstats/R/main.R

```diff
@@ -1,69 +1,289 @@
+# eegstats - main script: René  Freichel. Last updated on 19/05/2020.
 
-#' GA_Plot
-#'
-#' @description  The function \emph{GA_Plot} computes the grand average ERP signal for all conditions and plots it.
+# "This script contains several functions that will be the core part of the R package
+# eegstats. The first function (grand_average_plot) will plot the mean grand average ERP
+# signal over time. The second function allows the user to plot single - subject ERPs.
+
+# Testing
+# # For this purpose, we will use the example dataset ERPdata from the package erp.easy.
+# library(erp.easy)
+# my_erpdata <- ERPdata  # ERPdata from erp.easy will be used
+# View(my_erpdata) # subject x stimulus x time x electrodes
+
+
+# 1. ind_average_erp
+#' ind_average_erp
+#' @description  The function \emph{ind_average_erp} computes the single subject ERP signal for all conditions and a given electrode.
 #' @param my_erpdata Input Matrix with shape: subject x condition x time x electrodes.
 #' @param electrode A numeric value that specifies the number of electrode that should be plotted
-#'
-#'
-
+#' @param subject A character or a vector of characters that contains the number of subject (e.g. 'SO1')
 #' @details For more details, see Wikipedie page: \href{https://sccn.ucsd.edu/wiki/I.6:_Data_Averaging}{article}
+#' @return A matrix that contains the single-subject ERP for all conditions in a selected electrode. The matrix will be of shape condition x subject.
+#' @export ind_average_erp_data
+#' @seealso Add example of similar functions [still to be done].
+
+ind_average_erp <- function(my_erpdata, electrode = 6, subject = 'SO1') {
+
+  cond <- unique(my_erpdata[,2])
+  time <- unique(my_erpdata[,3])
+
+  # compute grand average (GA):
+  ind_average_erp_data <- array(NA, dim = c(length(subject),length(cond),length(time)));
```

Comment:
Just a suggestion (feel free to disagree), I think some of your new functions and objects would be easier to grasp if you renamed them slightly so that you first mention what they are and then the specifics, e.g.,
compute_ERP_XYZ
data_ERSP_XYZ
plot_XYZ

Especially because ERP and ERSP are so similar sounding, it gets a little confusing. Also, for end-users, it could become a bit tedious, as they can't use autocompletion with most functions/objects sharing the same beginning:
Grand_average_erp
Grand_average_ersp
Grand_average_erp_data
Grand_average_erp_plot